201-decrypt-running-windows-vm-without-aad/azuredeploy.json

```diff
@@ -0,0 +1,50 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vmName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the virtual machine"
+      }
+    },
+    "volumeType": {
+      "type": "string",
+      "defaultValue": "All",
+      "metadata": {
+        "description": "Type of the volume OS or Data to perform encryption operation"
+      }
+    },
+    "forceUpdateTag": {
+      "type": "string",
+      "defaultValue": "1.0",
+      "metadata": {
+        "description": "Pass in an unique value like a GUID everytime the operation needs to be force run"
+      }
+    }
+  },
+  "variables": {
+    "extensionName": "AzureDiskEncryption",
+    "extensionVersion": "2.2",
+    "encryptionOperation": "DisableEncryption"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Compute/virtualMachines/extensions",
+      "name": "[concat(parameters('vmName'),'/', variables('extensionName'))]",
+      "location": "[resourceGroup().location]",
+      "apiVersion": "2017-03-30",
+      "properties": {
+        "publisher": "Microsoft.Azure.Security",
+        "type": "[variables('extensionName')]",
+        "typeHandlerVersion": "[variables('extensionVersion')]",
+        "autoUpgradeMinorVersion": true,
+        "forceUpdateTag": "[parameters('forceUpdateTag')]",

```

Comment:
This shouldn't be needed (s/b an idempotent operation) - can you confirm?

Reply:
This is required in cases where user attaches additional data disks and wants  to perform encryption operations. In this case, CRP & GuestAgent won't invoke ADE extension unless there is a change in config. ForceUpdateTag is used as a mechanism to trigger re-running extension though there is no change in the extension config.

Reply:
ok, then let's add a defaultValue to the parameter of:

[uniqueString(deployment().name)]

Reply:
Portal, marketplace and the scripts in the repo will provide a timestamped deployment name so will work in the scenarios appropriate for the repo.  It's also just a default, it can be changed.